src/Domain/CoordinateMaps/TimeDependent/CubicScale.cpp

```diff
@@ -415,25 +411,6 @@ bool operator==(const CubicScale<Dim>& lhs,
 
 #define INSTANTIATE(_, data)                                                 \
   template class CubicScale<DIM(data)>;                                      \
-  template std::optional<                                                    \
```

Comment:
Don't we still need the instantiations for each `DIM`?

Reply:
I think instantiating the class for each DIM (`template class CubicScale<DIM(data)>;` on new line 413)  is enough to instantiate `inverse()`, since `inverse()` is no longer a member function template.  If it's not enough, the tests will fail.

Reply:
I tried adding such an explicit instantiation:

  CubicScale<DIM(data)>::inverse(                                            \
      const std::array<double, DIM(data)>& target_coords, const double time, \
      const std::unordered_map<                                              \
          std::string,                                                       \
          std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&         \
          functions_of_time) const noexcept;                                 \

but I got this compiler error:

/home/geoffrey/Codes/spectre/spectre/src/Domain/CoordinateMaps/TimeDependent/CubicScale.cpp:424:25: error: duplicate explicit instantiation of 'inverse'
GENERATE_INSTANTIATIONS(INSTANTIATE, (1, 2, 3))
                        ^
/home/geoffrey/Codes/spectre/spectre/src/Domain/CoordinateMaps/TimeDependent/CubicScale.cpp:424:25: note: previous explicit instantiation is here
/home/geoffrey/Codes/spectre/spectre/src/Domain/CoordinateMaps/TimeDependent/CubicScale.cpp:424:25: error: duplicate explicit instantiation of 'inverse'
GENERATE_INSTANTIATIONS(INSTANTIATE, (1, 2, 3))
                        ^
/home/geoffrey/Codes/spectre/spectre/src/Domain/CoordinateMaps/TimeDependent/CubicScale.cpp:424:25: note: previous explicit instantiation is here
/home/geoffrey/Codes/spectre/spectre/src/Domain/CoordinateMaps/TimeDependent/CubicScale.cpp:424:25: error: duplicate explicit instantiation of 'inverse'
GENERATE_INSTANTIATIONS(INSTANTIATE, (1, 2, 3))

I got no compiler error, and the tests all passed, when I removed this. Could you please clarify why we need it? Do you know why I was getting that compiler error?

Reply:
Ah, I see -- Mark is correct; the `DIM` is a template parameter of the class, not the function, so we only need the explicit instantiation for the class.